rspec-tools/rspec_tools/validation/description.py

```diff
@@ -16,8 +16,26 @@ def parse_names(path):
   section_names_path = read_file(path)
   return [s.replace('* ', '').strip() for s in section_names_path if s.strip()]
 
+def parse_security_standard_links(descr):
+  link_nodes = descr.find_all('a')
+  security_standards_links: Dict[str: List] = {}
+  for node in link_nodes:
+    href = node.attrs['href']
+    for standard, url_pattern in SECURITY_STANDARD_URL.items():
+      result = re.match(url_pattern, href)
+      if result is not None:
+        category = f"A{result[1].lstrip('0')}"
+        if standard not in security_standards_links.keys():
+          security_standards_links[standard] = []
+        security_standards_links[standard].append(category)
+  return security_standards_links
+
 HOW_TO_FIX_IT = 'How to fix it'
 HOW_TO_FIX_IT_REGEX = re.compile(HOW_TO_FIX_IT)
+SECURITY_STANDARD_URL = {
+  "OWASP": "https://(?:www\.)?owasp\.org/www-project-top-ten/2017/A(10|[1-9])_2017-",
+  "OWASP Top 10 2021": "https://(?:www\.)?owasp\.org/Top10/A(10|0[1-9])_2021-"
```

Comment:
This data is needed to accommodate the per-standard category formatting:
```suggestion
  "OWASP": {
    "url_pattern": r"https://(?:www\.)?owasp\.org/www-project-top-ten/2017/A(10|[1-9])_2017-",
    "convert_id": lambda value: f"A{value.lstrip('0')}",
  },
  "OWASP Top 10 2021": {
    "url_pattern": r"https://(?:www\.)?owasp\.org/Top10/A(10|0[1-9])_2021-",
    "convert_id": lambda value: f"A{value.lstrip('0')}",
  },
```
It also fixes an issue where the regexes need to be `r"..."` strings so that the `\.` escapes work correctly.